test/Microsoft.Azure.Devices.Edge.Test/helpers/ManualProvisioningFixture.cs

```diff
@@ -7,6 +7,90 @@ namespace Microsoft.Azure.Devices.Edge.Test.Helpers
     using Microsoft.Azure.Devices.Edge.Test.Common;
     using NUnit.Framework;
 
+    public class X509ManualProvisioningFixture : ManualProvisioningFixture
+    {
+        public X509Thumbprint Thumbprint { get; set; }
+        public EdgeDevice Device { get; set; }
+
+        public X509ManualProvisioningFixture()
+            : base()
+        {
+        }
+
+        [OneTimeSetUp]
+        public async Task X509ProvisionEdgeAsync()
+        {
+            await Profiler.Run(
+                async () =>
+                {
+                    using (var cts = new CancellationTokenSource(Context.Current.SetupTimeout))
+                    {
+                        CancellationToken token = cts.Token;
+                        DateTime startTime = DateTime.Now;
+
+                        this.Thumbprint = this.CreateSelfSignedCertificateThumbprint();
+
+                        EdgeDevice device = await EdgeDevice.GetOrCreateIdentityAsync(
+                        Context.Current.DeviceId + "-x509",
+                        this.iotHub,
+                        AuthenticationType.SelfSigned,
+                        this.Thumbprint,
+                        token);
+
+                        Context.Current.DeleteList.TryAdd(device.Id, device);
+
+                        this.Device = device;
+
+                        await this.ManuallyProvisionEdgeAsync(device, startTime, token);
+                    }
+                },
+                "Completed edge manual provisioning");
+        }
+
+        private X509Thumbprint CreateSelfSignedCertificateThumbprint()
+        {
+            return new X509Thumbprint()
+            {
+                PrimaryThumbprint = "9991572f0a02bdc7c89fc032b95d79aca18ef7a3",
+                SecondaryThumbprint = "9991572f0a02bdc7c89fc032b95d79aca18ef7a4"
```

Comment:
1. How are these Thumbprint generated? Could you add an instruction on how to maintain these thrumbprints in case of renewal. 
2. Usually the secrets are fed into an E2E using either `config.yaml` or scriptically queried using Key Vault